### DISCUSSION TITLE
Redefine the profiles structure

### DISCUSSION THREAD

**Discussion by datejada** [TulipaEnergy/TulipaEnergyModel.jl#517] [Ideas]
The current data in the `assets-profile.csv` and `flows-profiles.csv` files assumes that all profiles are associated with availability profiles, which is true in most cases. Still, for energy storage, the profile is commonly associated with the energy inflows the asset can get. There are other potential applications for profiles, such as defining a minimum production profile (e.g., nuclear) or a minimum or maximum storage level (e.g., hydro). In all these cases, we will need to determine what type of profile is and then use it in the model constraints. Also, the profiles can be defined for inter-temporal constraints (i.e., between representative periods for the seasonal storage) or intra-temporal constraints (i.e., within the representative periods).

To solve this, there are different options that I have thought (I'm open to other options):

1. Add a profile `id` column to the files. This will allow us to have several profiles associated with an `asset` or flow. Then we can have a new file, `profile-definition.csv`, to define the profile type (e.g., availability, inflows, minimum/maximum storage level, etc.). Here, maybe the profiles become a structure that has as a field its type (I'm not sure)
2. Idem as 1, but instead of creating an id in a new file, the current profiles files have a new column with the definition. Then, we have the tuple (asset, profile_type) to identify each profile type.
3. Idem as 1, but creating a new structure in the model for the profiles (currently, the profiles are defined within the asset and flow structures)

Any other ideas? @suvayu @abelsiqueira @gnawin @clizbe @g-moralesespana 

Suvayu could have a better overview of what is better since he is with the IO.

**Comment by abelsiqueira** [comment]
Knowing how the data will come from the InfluxDB (EDR?) helps. So we don't deviate more than necessary.

Assuming something like #505 (using dataframes for everything) gets implemented, then it could be just another column in the profiles column.

**Comment by abelsiqueira** [comment]
We have decided what to do, and will update the issue with the closing action
